x-pack/plugins/cases/public/components/case_view/case_view_metrics.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { EuiFlexGroup, EuiFlexItem, EuiLoadingSpinner, EuiPanel } from '@elastic/eui';
+import { CaseMetrics } from '../../../common/ui';
+import {
+  ASSOCIATED_HOSTS_METRIC,
+  ASSOCIATED_USERS_METRIC,
+  TOTAL_ALERTS_METRIC,
+  TOTAL_CONNECTORS_METRIC,
+} from './translations';
+
+const MetricValue = styled(EuiFlexItem)`
+  font-size: ${({ theme }) => theme.eui.euiSizeL};
+  font-weight: bold;
+`;
+
+export interface CaseViewMetricsProps {
+  metrics: CaseMetrics | null;
+  isLoading: boolean;
+}
+
+type MetricItems = Array<{ title: string; value: number }>;
+
+const useMetricItems = (metrics: CaseMetrics | null): MetricItems => {
+  const { alerts, connectors } = metrics ?? {};
+  const totalConnectors = connectors?.length;
+  const alertsCount = alerts?.count;
+  const totalAlertUsers = alerts?.users?.total;
+  const totalAlertHosts = alerts?.hosts?.total;
```

Comment:
Maybe it is better to default them to `0` if they are `undefined`. What do you think?

Reply:
I believe we're leveraging whether they're undefined here to determine whether or not to actually display the metric in the UI. I think that means we're relying on the backend to determine what is shown in the UI. I'm ok with either approach but maybe it'd make more sense to loop over the `features.metrics` array to build the components and default them to `0` if the backend for some reason didn't return them.